packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/namespaces/_components/formNamespace/useNamespaceForm.hook.tsx

```diff
@@ -63,18 +65,44 @@ export const useNamespaceForm = ({
 
   const typeRules = z.nativeEnum(database.m3db.namespace.TypeEnum);
 
-  const schema = z.object({
-    name: nameRules,
-    resolution: mandatoryShortTimeRules,
-    blockDataExpirationDuration: optionalShortTimeRules,
-    blockSizeDuration: optionalShortTimeRules,
-    bufferFutureDuration: optionalShortTimeRules,
-    bufferPastDuration: optionalShortTimeRules,
-    periodDuration: mandatoryShortTimeRules,
-    snapshotEnabled: z.boolean().optional(),
-    type: typeRules,
-    writesToCommitLogEnabled: z.boolean().optional(),
-  });
+  const schema = z
+    .object({
+      name: nameRules,
+      periodDuration: mandatoryShortTimeRules,
+      resolution: optionalShortTimeRules,
+      blockDataExpirationDuration: optionalShortTimeRules,
+      blockSizeDuration: optionalShortTimeRules,
+      bufferFutureDuration: optionalShortTimeRules,
+      bufferPastDuration: optionalShortTimeRules,
+      snapshotEnabled: z.boolean().optional(),
+      type: typeRules,
+      writesToCommitLogEnabled: z.boolean().optional(),
+    })
+    .superRefine((values, context) => {
+      if (
+        values.type === database.m3db.namespace.TypeEnum.aggregated &&
+        !values.resolution
+      ) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: t('formNamespaceErrorMinLength', {
+            min: NAMESPACES_CONFIG.name.min,
+          }),
+          path: ['resolution'],
+        });
+      }
+      if (
+        values.type === database.m3db.namespace.TypeEnum.aggregated &&
+        durationStringToSeconds(values.periodDuration) <
+          durationStringToSeconds(values.resolution)
+      ) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: t('formNamespaceDurationResolutionErrorPattern'),
+          path: ['resolution'],
+        });
```

Comment:
The `context.AddIssue` calls could be refactored in a single function that takes a `message` as argument, to prevent from code repetition.

Reply:
yes, thanks, it's fixed